File: Leetcode/HashTables/sortColors.ts
// 75. Sort Colors
// https://leetcode.com/problems/sort-colors/
/**
 Do not return anything, modify nums in-place instead.
 */
function sortColors(nums: number[]): void {
    let colorCount: { [key: number]: number } = {};

    for (let num of nums) {
        colorCount[num] = (parseInt(colorCount[num]) || 0) + 1;
    }

    let i: number = 0;
    console.log(colorCount)
    for (let color in colorCount) {
        while (colorCount[color] > 0) {
            nums[i] = color;
            i++;
            colorCount[color]--;
        }
    }
};

File: Leetcode/HashTables/sortColors.py
# 75. Sort Colors
# https://leetcode.com/problems/sort-colors/
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        dict = {
            0: 0,
            1: 0,
            2: 0
        }

        for num in nums:
            dict[num] += 1
        i = 0
        for color in dict:
            while dict[color] > 0:
                nums[i] = color
                i += 1
                dict[color] -= 1

# Time: O(2n) = O(n)
# Space: O(1)
